### ISSUE TITLE
error 1

### ISSUE THREAD

**Issue by jayheavner** [sonos-web/sonos-web#6]
Hi, I just installed per the instructions, on a Windows 10 box. sonos-web -V reports 1.3.0. It gives no information, just says error 1 when it goes to start. I tried doing an update but the same thing happened.

**Comment by Villarrealized** [comment]
Did the installer run fine? There are several logs files that will be located at ~/.sonos-web. They may be of use in debugging.

I should be able to test it on a Windows 10 machine in the next day or two.

**Comment by jayheavner** [comment]
Yeah, the installer ran fine. I had looked at the install logs and didn't see anything weird. Couple of warnings. I'm thinking this might be an issue with my global install directory. Win10 seems wildly inconsistent with where it installs packages based on the console being used. PowerShell seems to want to install a different place than a regular command prompt and I am bad about being consistent in which I'm using. I did manually start the server component and it came up fine. I'll dig a bit as I have time.

**Comment by Villarrealized** [comment]
Hmm okay, yes sounds like a pathing issue. I will try to test it soon myself.

**Comment by jayheavner** [comment]
I threw in some debugging.

`- Starting Sonos Web----------------------------------------
command > C:\Users\jheavner\AppData\Roaming\npm\node_modules\sonos-web-cli\node_modules\forever/bin/forever --pidFile C:\Users\jheavner/.sonos-web/sonos-web.pid -a -l C:\Users\jheavner/.sonos-web/forever.log -o C:\Users\jheavner/.sonos-web/sonos-web.log -e C:\Users\jheavner/.sonos-web/sonos-web.log start src/server.js
----------------------------------------

----------------------------------------
code > 1
stdout >
stderr > 'C:\Users\jheavner\AppData\Roaming\npm\node_modules\sonos-web-cli\node_modules\forever/bin/forever' is not recognized as an internal or external command,
operable program or batch file.

----------------------------------------`

It's clearly something weird with the forever package. The file exists at the location referenced. I went ahead and installed forever globally and hardcoded the path to the global install with the same error.  A quick Google shows issues with forever on Windows but I didn't really dig into it.

If I directly run the src/server.js everything works as expected. That's easy enough.

Windows....sigh.

**Comment by Villarrealized** [comment]
Hmm, interesting for sure. That's unfortunate. I tried installing on Windows 10 through cmd.exe and my installation failed on installing back-end dependencies. Haven't dug into it yet. Glad you were able to get it working manually though.

**Comment by jellmers** [comment]
Hi,
I am getting the same issue on Windows10.  And not really much of a clue how to resolve it :-(

Any pointers?

**Comment by gieljnssns** [comment]
When I try to update Sonos Web, I see error 1
```
pi@MQTT-PiHole-Pi:~ $ sonos-web update
Stopping Sonos Web if it is running...
✖ Stopping Sonos Web
error 1
✔ Uninstalling Sonos Web
Installing Sonos Web
✔ downloading installation files
✔ installing front-end dependencies
✔ build front-end application
✔ install back-end dependencies
✔ cleaning up installation files
✔ Starting Sonos Web

success Open a web browser on this computer to http://localhost:5050 to start!
You can access Sonos Web network-wide by going to http://192.168.79.35:5050
```
install.log:
```
pi@MQTT-PiHole-Pi:~/.sonos-web $ cat install.log

> nodent-runtime@3.2.1 install /home/pi/.sonos-web/client/node_modules/nodent-runtime
> node build.js

## Built /home/pi/.sonos-web/client/node_modules/nodent-runtime/dist/index.js

> yorkie@2.0.0 install /home/pi/.sonos-web/client/node_modules/yorkie
> node bin/install.js

setting up Git hooks
can't find .git directory, skipping Git hooks installation
added 1339 packages from 1098 contributors and audited 27117 packages in 148.385s
found 18 vulnerabilities (10 moderate, 8 high)
  run `npm audit fix` to fix them, or `npm audit` for details
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})


> sonos-web@0.9.4 build /home/pi/.sonos-web/client
> vue-cli-service build


 WARNING  Compiled with 5 warnings10:18:05 AM

Module Warning (from ./node_modules/eslint-loader/index.js):
error: 'response' is never reassigned. Use 'const' instead (prefer-const) at src/views/services/Spotify.vue:50:11:
  48 |   methods: {
  49 |     async authorize() {
> 50 |       let response = await SpotifyAPI.getAuthURL();
     |           ^
  51 |       window.location = response.data.authURL;
  52 |     },
  53 |   },


1 error found.
1 error potentially fixable with the `--fix` option.

Module Warning (from ./node_modules/eslint-loader/index.js):
error: Unexpected console statement (no-console) at src/views/library/Playlists.vue:46:7:
  44 |       this.loading = false;
  45 |       this.playlists = deepmerge(this.playlists, data);
> 46 |       console.log(this.playlists);
     |       ^
  47 |     },
  48 |     loadingError(error) {
  49 |       this.loading = false;


1 error found.

You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.
 warning  in ./src/mixins/spotifyAuth.js

Module Warning (from ./node_modules/thread-loader/dist/cjs.js):

/home/pi/.sonos-web/client/src/mixins/spotifyAuth.js
  16:9  error  Unexpected console statement  no-console
  27:9  error  Unexpected console statement  no-console

✖ 2 problems (2 errors, 0 warnings)


 @ ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/vuetify-loader/lib/loader.js!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/views/spotify/SpotifyArtist.vue?vue&type=script&lang=js& 32:0-47 39:11-22
 @ ./src/views/spotify/SpotifyArtist.vue?vue&type=script&lang=js&
 @ ./src/views/spotify/SpotifyArtist.vue
 @ ./src/router.js
 @ ./src/main.js
 @ multi ./src/main.js

 warning

asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  js/chunk-vendors.dc27736d.js (398 KiB)

 warning

entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  app (585 KiB)
      css/chunk-vendors.98b019a3.css
      js/chunk-vendors.dc27736d.js
      css/app.ac07faf7.css
      js/app.f8da5b48.js


  File                                      Size             Gzipped

  dist/js/chunk-vendors.dc27736d.js         397.89 KiB       127.14 KiB
  dist/js/app.f8da5b48.js                   39.66 KiB        11.69 KiB
  dist/js/Play Queue.ad15e106.js            36.56 KiB        10.88 KiB
  dist/js/Album.46b2832b.js                 25.91 KiB        8.24 KiB
  dist/js/Play Queue~Rooms.3deebb9b.js      23.99 KiB        8.68 KiB
  dist/js/Spotify Album.0382fd8c.js         20.16 KiB        6.38 KiB
  dist/js/SearchTopResults.0ae25cb4.js      19.16 KiB        5.29 KiB
  dist/js/Songs.4f4bb069.js                 16.25 KiB        5.61 KiB
  dist/js/Spotify Artist.d9d4e4cc.js        16.02 KiB        5.23 KiB
  dist/js/Spotify Songs.c29cea35.js         15.77 KiB        5.57 KiB
  dist/js/Sonos Favorites.7c6913a5.js       15.63 KiB        5.38 KiB
  dist/js/Genre.ed923e02.js                 15.19 KiB        5.05 KiB
  dist/js/Artist.0c869650.js                15.12 KiB        5.06 KiB
  dist/js/Share.4a0bc857.js                 15.10 KiB        5.03 KiB
  dist/js/Music Library~My                  14.66 KiB        4.43 KiB
  Sonos~Search~Spotify.9ce3ffd9.js
  dist/js/Sonos Playlists.b83d9c1b.js       13.68 KiB        4.63 KiB
  dist/js/Playlists.defcff10.js             13.66 KiB        4.63 KiB
  dist/js/Artists.d6c66893.js               13.64 KiB        4.60 KiB
  dist/js/Shares.8bbe9732.js                13.59 KiB        4.61 KiB
  dist/js/Genres.6505890c.js                13.59 KiB        4.59 KiB
  dist/js/Albums.c049383a.js                13.59 KiB        4.60 KiB
  dist/js/Spotify Playlists.534e3d3f.js     13.20 KiB        4.61 KiB
  dist/js/Spotify Albums.802b8090.js        13.13 KiB        4.61 KiB
  dist/js/Rooms.ee82a8f5.js                 11.20 KiB        3.69 KiB
  dist/js/Now Playing.1e0de58f.js           4.90 KiB         1.69 KiB
  dist/js/Spotify.e93c7717.js               3.82 KiB         1.54 KiB
  dist/js/Search.f763c4fa.js                3.30 KiB         1.37 KiB
  dist/js/Music Library.af00731d.js         1.16 KiB         0.64 KiB
  dist/js/My Sonos.6e8e6e2c.js              0.98 KiB         0.59 KiB
  dist/css/chunk-vendors.98b019a3.css       141.13 KiB       18.63 KiB
  dist/css/Play Queue.a362c20d.css          19.35 KiB        3.18 KiB
  dist/css/app.ac07faf7.css                 6.66 KiB         1.84 KiB
  dist/css/Music Library~My                 5.78 KiB         1.19 KiB
  Sonos~Search~Spotify.39e8e919.css
  dist/css/Rooms.2aba62ab.css               3.54 KiB         1.07 KiB
  dist/css/Play Queue~Rooms.25b719f9.css    3.28 KiB         0.85 KiB
  dist/css/SearchTopResults.411e26a0.css    1.24 KiB         0.46 KiB
  dist/css/Spotify Songs.7a6cccbc.css       0.90 KiB         0.32 KiB
  dist/css/Songs.7a6cccbc.css               0.90 KiB         0.32 KiB
  dist/css/Sonos Favorites.bd81b72e.css     0.90 KiB         0.32 KiB
  dist/css/Spotify Album.7a6cccbc.css       0.90 KiB         0.32 KiB
  dist/css/Album.7a6cccbc.css               0.90 KiB         0.32 KiB
  dist/css/Search.6681b166.css              0.45 KiB         0.25 KiB
  dist/css/Share.ba7a0aba.css               0.33 KiB         0.22 KiB
  dist/css/Genre.5ff8726c.css               0.31 KiB         0.22 KiB
  dist/css/Sonos Playlists.afbb6df5.css     0.24 KiB         0.17 KiB
  dist/css/Artist.afbb6df5.css              0.24 KiB         0.17 KiB
  dist/css/Shares.afbb6df5.css              0.24 KiB         0.17 KiB
  dist/css/Genres.afbb6df5.css              0.24 KiB         0.17 KiB
  dist/css/Spotify                          0.24 KiB         0.17 KiB
  Playlists.afbb6df5.css
  dist/css/Playlists.afbb6df5.css           0.24 KiB         0.17 KiB
  dist/css/Albums.afbb6df5.css              0.24 KiB         0.17 KiB
  dist/css/Artists.afbb6df5.css             0.24 KiB         0.17 KiB
  dist/css/Spotify Albums.afbb6df5.css      0.24 KiB         0.17 KiB
  dist/css/Spotify Artist.afbb6df5.css      0.24 KiB         0.17 KiB
  dist/css/Now Playing.48d3be0e.css         0.20 KiB         0.15 KiB

  Images and other types of assets omitted.

 DONE  Build complete. The dist directory is ready to be deployed.
 INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html

-  Building for production...
added 157 packages from 171 contributors and audited 2872 packages in 18.419s
found 3 vulnerabilities (1 moderate, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
info:    Forever processing file: src/server.js
```
forever.log:
```
pi@MQTT-PiHole-Pi:~/.sonos-web $ cat forever.log
{ clientId: 'd80589162xxxxxxxxxxxxxxxxc6d69ae46',
  clientSecret: '2836db0cbxxxxxxxxxxxxxxxxx65730a35',
  redirectUri: 'http://localhost:5050/spotify' }
Server listening on port 5050
```
Sonos-web.log:
```
pi@MQTT-PiHole-Pi:~/.sonos-web $ cat sonos-web.log
{ clientId: 'd80589162xxxxxxxxxxxxxxxxc6d69ae46',
  clientSecret: '2836db0cbxxxxxxxxxxxxxxxxx65730a35',
  redirectUri: 'http://localhost:5050/spotify' }
Server listening on port 5050
```

**Comment by ayust** [comment]
I was getting `error 1` when trying to run `sonos-web start` on Windows 10. I was able to fix it by modifying the cli script to invoke `node ${foreverPath} ...` rather than just `${foreverPath}` - guessing this has to do with Windows not honoring the shebang in forever's startup script.

**Comment by NyquistFX** [comment]
> I was getting `error 1` when trying to run `sonos-web start` on Windows 10. I was able to fix it by modifying the cli script to invoke `node ${foreverPath} ...` rather than just `${foreverPath}` - guessing this has to do with Windows not honoring the shebang in forever's startup script.

Ayust, could you elaborate exactly what you changed and where please?

I am getting the same error despite trying a variety of things - my thoughts are something to do with conflicts on port 5050 which seems to be tied up by "Connected Devices Platform Service" - but even with this service shutdown and disabled the same error occurs.

Running NETSTAT does still show a process with UDP open on that port, however when you try to kill the associated PID from the command line it tells you that process doesn't exist.

I tried to find where I might change the sonos-web port rather than getting rid of conflicting services but came up empty in that regard as well.

Any further pointers appreciated..

**Comment by Villarrealized** [comment]
@NyquistFX I'm not sure your error has to do with the port, but if you'd like to try changing it you can do so in  `~/.sonos-web/.env`

**Comment by Villarrealized** [comment]
This issue should be fixed with v1.3.1 of `sonos-web-cli`. Run `npm update -g sonos-web-cli`
